Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -141,46 +141,38 @@ public static Waveform CreateDetroughEnvelopeWaveform(Waveform referenceWaveform
             Waveform envelopeWaveform = CloneAndConditionReferenceWaveform(referenceWaveform);
             WritableBuffer<ComplexSingle> envWfmWriteBuffer = envelopeWaveform.Data.GetWritableBuffer();
 
-            double[] iqMagnitudes = referenceWaveform.Data.GetMagnitudeDataArray(false);
+            double[] envelope = envelopeWaveform.Data.GetMagnitudeDataArray(false);
             double detroughRatio = detroughConfig.MinimumVoltage_V / detroughConfig.MaximumVoltage_V;
 
             // Waveforms are assumed to be normalized in range [0, 1], so no normalization will happen here
             switch (detroughConfig.Type)
             {
                 case DetroughType.Exponential:
                     // Formula: e = IQmag + d*exp(-IQmag/d)
-                    double expScale = 1 + detroughRatio * Math.Exp(-1.0 / detroughRatio);
                     for (int i = 0; i < envWfmWriteBuffer.Size; i++)
-                    {
-                        double sampleValue = iqMagnitudes[i] + detroughRatio * Math.Exp(-iqMagnitudes[i] / detroughRatio);
-                        // Scale detrough to Vmax. Divide by detroughed waveform's max value to normalize. IQMagnitude's max value is 1
-                        envWfmWriteBuffer[i] = ComplexSingle.FromSingle((float)(detroughConfig.MaximumVoltage_V * sampleValue / expScale));
-                    }
+                        envelope[i] = envelope[i] + detroughRatio * Math.Exp(-envelope[i] / detroughRatio);
                     break;
                 case DetroughType.Cosine:
                     // Formula: e = 1 - (1-d)*cos(IQmag*pi/2)
-                    double cosScale = 1 - (1 - detroughRatio) * Math.Cos(Math.PI / 2);
                     for (int i = 0; i < envWfmWriteBuffer.Size; i++)
-                    {
-                        double sampleValue = 1 - (1 - detroughRatio) * Math.Cos(iqMagnitudes[i] * cosScale);

```

Comment:
This is likely the culprit of the issues. _cosScale_ is not used correctly here. Based on the formula it ought to be equal to pi/2, but is definitely not.